overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,96 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// StoreConfigSnapshotMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+// The caller is responsible for locking the state.
+func StoreConfigSnapshotMaybe(st *state.State, snapName string, rev snap.Revision) error {
+	var config map[string]interface{}                     // snap => configuration
+	var configSnapshots map[string]map[string]interface{} // snap => revision => configuration
```

Comment:
If the above sounds reasonable, this might be `revisionConfig`.. the most interesting fact about that map is that it stores the configuration for particular snap revisions, so calling that out in the name feels like a win.

Also, per our conversation in the call last week, we might benefit from not unmarshaling the configuration of either of those. We're simply moving the blob around, so we can use `json.RawMessage` in both cases, and then assign one to the other when necessary.

Note that there's a bug in the json pacakge when handling those without them being a pointer, so we probably want something like:
```
var config map[string]*json.RawMessage
var revisionConfig map[string]map[string]*json.RawMessage
```